### ISSUE TITLE
Latest NMS App version (2.19.3) is crashing on iPhone 7

### ISSUE THREAD

**Issue by albertdick** [AssistantNMS/App#235]
### 📜 Describe the bug

Today, December 3rd, 2023, I updated the iOS app to version 2.19.3 and after the update I can no longer run it. When opening the app, it crashes immediately.

I've already tried uninstalling and reinstalling the app, but it didn't work.

I'm using iOS version 15.8 on iPhone 7.

### 👟 To Reproduce

When opening the app, it crashes.

### 👍 Expected behavior

That when opening the app, it works. :)

### 📱 Where did you find this problem?

iOS

### 👀 Have you checked if this bug report has been created already?

- [X] I checked and couldn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/AssistantNMS/App/blob/main/.github/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by Khaoz-Topsy** [comment]
Hey @albertdick ! 
Thank you for creating this issue, I appreciate the details you have included! 

I am back from my holiday and hopeful that I will get this issue sorted 😅. I am starting the release of a new version now, it should be available in a day if the Apple review process goes smoothly. Once it is live, would you be able to test it for me? If you are willing to help in this I will let you know about it via a comment on this issue 😋

**Comment by albertdick** [comment]
Of course. I'll wait your instructions.
Thanks!

**Comment by Khaoz-Topsy** [comment]
Thank you so much! The update should be available now 😁
Please test this update for me when you have the time and let me know if it is now working 😋

**Comment by albertdick** [comment]
I can't install a standalone IPA app on my iPhone. I can only install those available on the App Store. Is it possible for me to test it with TestFlight or some other method without a jailbreak?

**Comment by Khaoz-Topsy** [comment]
Sorry for the confusion the update is live on the App Store, you should just be able to update the app to the newest version 2.20.0 to test it 😁. If this version doesn't work I will invite you to the app's TestFlight version for faster testing 😋. 

I needed to update the data in the app so I did a full release for Android and iOS as well as hopefully resolve the dependency issue in the app on iOS 16 devices 😋

**Comment by albertdick** [comment]
Unfortunately v2.20.0 didn't solve the issue.
The app continues to crash at opening.
Here is a reference video.

https://github.com/AssistantNMS/App/assets/8559840/fc93a35c-8b28-40bd-a7c0-323759ec4731

**Comment by Khaoz-Topsy** [comment]
Damn, that is not what I was expecting 😕. In XCode I have set the app to support iOS 11 and up, there must be some package that I am using that is expecting something to work that is only on iOS 17 🤔

Can you try uninstalling the app, then joining the TestFlight account (https://testflight.apple.com/join/zoV1ghU1). I am hoping that TestFlight will at least give me the crash reports so I can do something about what is breaking instead of guessing what the problem is 😅. The link to join my TestFlight says that the beta isn't accepting new testers at the moment, but that should be sorted in 30min once apple does the automatic scan of the app bundle again. I will check it again later and let you know when the link decides to work again 😅

---

If you are feeling extra generous with your time, could you try getting the logs of the app? According to an Apple article (https://developer.apple.com/documentation/xcode/acquiring-crash-reports-and-diagnostic-logs) these are the steps:

1. Open the Analytics & Improvements section of Settings on the device. See [Share analytics, diagnostics, and usage information with Apple](https://support.apple.com/en-us/HT202100).

0. Tap Analytics Data.

0. Locate the log for your app. The log name starts with <AppBinaryName>_<DateTime> for crash reports, or JetsamEvent_<DateTime> for high-memory use crashes.
   - Should have `assistantnms_app`, `com.kurtlourens.assistantnms_app`, `com.kurtlourens.nomansskyrecipes` or something with `kurtlourens` in the name. Or just look at the date and time it happened 😅

0. Select the desired log.

0. Tap the Share icon, and select Mail to send the crash report as a mail attachment.

   - You can use support@nmsassistant.com


---


Thanks again for the help!

**Comment by Khaoz-Topsy** [comment]
Still waiting, at this rate it would be faster to do a full release 😅
![image](https://github.com/AssistantNMS/App/assets/15005470/1d79741d-f935-4e31-a9c8-5499b08bbaee)

**Comment by Khaoz-Topsy** [comment]
Hi again @albertdick! The TestFlight link to join (https://testflight.apple.com/join/zoV1ghU1) should be working now 😁. The version available on it is the currently live version, so the most we can hope for is some detailed logs about the crash from TestFlight... 

In the other version (2.20.1), I updated all of the dependencies. I am hopeful that 2.20.1 will contain a fix for whatever dependency is breaking 🤞. I will let you know when 2.20.1 has completed processing 😁

**Comment by albertdick** [comment]
Ok, I managed to install version 2.20.1 through TestFlight, and the app still crashes when opened. I sent feedback through TestFlight.

I couldn't find any crash logs in the "Analytics & Improvements" section of Settings device with the corresponding names (`assistantnms_app`, `com.kurtlourens.assistantnms_app`, `com.kurtlourens.nomansskyrecipes` or just `kurtlourens`), both for the App Store crash and the TestFlight crash.

I tried filtering the logs from today (2023-12-21), but I only have these matches.

awdd-2023-12-21-xxxxx
JetsamEvent-2023-12-21-xxxxxx
Runner-2023-12-21-xxxxxx
SiriSearchFeedBack-2023-12-21-xxxxx
spotlightknowledged.cpu_resouce-2023-12-21-xxxxx
xp_amp_app_usage_dnu-2023-12-21-xxxxx

**Comment by Khaoz-Topsy** [comment]
Hey @albertdick, sorry for being so quiet the last 2 weeks 😅. I am still working on fixing the app for iOS 16, I just have to work around my full-time job 😅. I did find a way to get the app crashing on an iOS16 simulator on a virtual Mac. From there it seems like the analytics is causing problems on iOS17 because of permissions issues. So I have made a new version without any of the analytics. I made the update to TestFlight if you want to try it out. I also tried to push this version to the App Store page but it has been rejected because the app has 3rd images in it 🙄... So I will resubmit and hopefully I will get a different Apple Employee that wont reject the app 😞
